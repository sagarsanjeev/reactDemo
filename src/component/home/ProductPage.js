import React from 'react';

const showLatest = ({latestData}) => {
    if(latestData){
        return <div className="row row-cols-1 row-cols-md-3 mb-3 text-center courseBox" style={{margin: '50px', paddingBottom: '80px'}}>
                {latestData && latestData.products.map((data) => {
                    return <div className="card" style={{width: '18rem', margin: '10px'}} key={data.Id}>
                                <img className="card-img-top" src={data.img} alt={data.Title}/>
                                <div className="card-body">
                                    <h5 className="card-title">{data.Title}</h5>
                                    <p className="card-text">{data.Description}</p>
                                    <a href={data.Url} className="btn btn-primary">Link</a>
                                </div>
                            </div>
                })}
                </div>
    }
}

const ProductPage = (props) => {
    return(
        <div className="home-latest">
            {showLatest(props)}
        </div>
    )
}

export default ProductPage;