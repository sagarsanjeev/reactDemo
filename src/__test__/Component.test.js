import React from 'react';
import ProductPage from '../component/home/ProductPage';
import {create} from 'react-test-renderer';

describe('Products Component Snapshot', () => {
    test("Products",() => {
        let tree = create(<ProductPage/>);
        expect(tree.toJSON()).toMatchSnapshot();
    })
})
