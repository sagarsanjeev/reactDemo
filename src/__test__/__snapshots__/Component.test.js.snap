// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component Snapshot Products 1`] = `
<div
  className="home-latest"
/>
`;
